definitions/checks/katello_agent.rb

```diff
@@ -0,0 +1,39 @@
+require 'yaml'
+
+class Checks::CheckKatelloAgentEnabled < ForemanMaintain::Check
+  metadata do
+    label :check_katello_agent_enabled
+    description 'Check whether the katello-agent feature is enabled before upgrading'
+    tags :pre_upgrade
+
+    confine do
+      feature(:instance).downstream
+    end
+  end
+
+  INSTALLER_ANSWERS_FILE = '/etc/foreman-installer/scenarios.d/satellite-answers.yaml'.freeze
+
+  def run
+    assert(
+      !katello_agent_enabled?,
+      "The katello-agent feature is enabled on this system. As of Satellite 6.15.0, katello-agent"\
+      " is removed and will no longer function. Before proceeding with the upgrade, you should"\
+      " ensure that you have deployed and configured an alternative tool for remote package"\
+      " management and patching for content hosts, such as Remote Execution (REX) with pull-based"\
+      " transport. See the Managing Hosts guide in the Satellite documentation for more info."\
+      " Disable katello-agent with the command"\
+      " `satellite-installer --foreman-proxy-content-enable-katello-agent false`"\
```

Comment:
I think our idea was to stop the upgrade _on purpose_, even though technically it may not be strictly required. This way the user would wake up and say, wait a minute, I need to make sure all my ducks are in a row for host patching before I upgrade!

Reply:
after all, our user story does say

"As a user with katello-agent enabled, I am prevented from upgrading to Satellite 6.15 / Katello 4.10 until I disable katello-agent" :)

Reply:
Mhh. Installer runs are rather expensive and I'd love to avoid them.

Would your story be also happy if we warn the user that they will lose the feature and they have to accept that by pressing "yes"?

Reply:
> Mhh. Installer runs are rather expensive and I'd love to avoid them.
> 
> Would your story be also happy if we warn the user that they will lose the feature and they have to accept that by pressing "yes"?

I think this is fine.

Reply:
I am curious to know what @jeremylenz thinks about this as well.

Reply:
An interactive choice seems fine as well. Let me research the best way to implement it.

Reply:
I am quite sure we had similar in the past, if you don't find something, I can have a look tomorrow morning

Reply:
Closing the loop: let's keep this the way it is, as discussed offline. Expensive installer run that adds a few minutes is better than being able to proceed thru an installation that breaks host patching unbeknownst to you.

Reply:
Do you want this to be skippable? Users can whitelist this step and get the automatic removal. That seems fine but I want to make sure that matches your desired behaviour.

Reply:
I added some language explaining what will happen if the user uses `--whitelist` to skip the check, which indeed should be the fastest way to proceed if they elect to do so.